.github/scripts/tag_docker_images_for_release.py

```diff
@@ -0,0 +1,48 @@
+import generate_binary_build_matrix
+import os
+import subprocess
+import argparse
+from typing import Dict
+
+def tag_image( image: str, default_tag: str, release_version: str, dry_run: str, tagged_images: Dict[str, bool]) -> None:
+    if image not in tagged_images:
+        release_image = image.replace(f'-{default_tag}', f'-{release_version}')
+        print(f"Tagging {image} to {release_image} , dry_run: {dry_run}")
+        if dry_run == 'disabled':
+            subprocess.check_call(["docker", "pull", image])
+            subprocess.check_call(["docker", "tag", image, release_image])
+            subprocess.check_call(["docker", "push", release_image])
+
+        tagged_images[image] = True
```

Comment:
Dismantle `if` pyramid a bit
```suggestion
    if image in tagged_images:
        return
    release_image = image.replace(f'-{default_tag}', f'-{release_version}')
    print(f"Tagging {image} to {release_image} , dry_run: {dry_run}")
    
    if dry_run == 'disabled':
        subprocess.check_call(["docker", "pull", image])
        subprocess.check_call(["docker", "tag", image, release_image])
        subprocess.check_call(["docker", "push", release_image])
    tagged_images[image] = True
```